Index: rainmaker/pt-services-v2/src/main/java/org/egov/pt/repository/rowmapper/PropertyRowMapper.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- rainmaker/pt-services-v2/src/main/java/org/egov/pt/repository/rowmapper/PropertyRowMapper.java	(revision da92d8583591bd7d8ad9d44f3160a52eb865b80d)
+++ rainmaker/pt-services-v2/src/main/java/org/egov/pt/repository/rowmapper/PropertyRowMapper.java	(revision c21a11b74318b8a0fcdd610ac16dbca8a1f0146c)
@@ -7,6 +7,7 @@
 import java.util.HashMap;
 import java.util.List;
 import java.util.Map;
+import java.util.Set;
 
 import org.egov.pt.web.models.*;
 import org.egov.pt.web.models.Property.CreationReasonEnum;
@@ -46,13 +47,18 @@
 						.longitude(rs.getDouble("longitude")).pincode(rs.getString("pincode"))
 						.street(rs.getString("street")).tenantId(tenanId).type(rs.getString("type")).build();
 
+				AuditDetails auditdetails = AuditDetails.builder().createdBy(rs.getString("createdBy"))
+						.createdTime(rs.getLong("createdTime")).lastModifiedBy(rs.getString("lastModifiedBy"))
+						.lastModifiedTime(rs.getLong("lastModifiedTime"))
+						.build();
+
 				currentProperty = Property.builder().address(address)
 						.acknowldgementNumber(rs.getString("acknowldgementNumber"))
 						.creationReason(CreationReasonEnum.fromValue(rs.getString("creationReason")))
 						.occupancyDate(rs.getLong("occupancyDate")).propertyId(currentId)
 						.oldPropertyId(rs.getString("oldPropertyId"))
 						.status(PropertyInfo.StatusEnum.fromValue(rs.getString("status")))
-						.tenantId(tenanId).build();
+						.tenantId(tenanId).auditDetails(auditdetails).build();
 
 				propertyMap.put(currentId, currentProperty);
 			}
@@ -64,6 +70,8 @@
 
 	private void addChildrenToProperty(ResultSet rs, Property property) throws SQLException {
 
+
+
 		PropertyDetail detail = PropertyDetail.builder()
 				.additionalDetails(rs.getObject("additionalDetails")).buildUpArea(rs.getFloat("buildUpArea"))
 				.channel(ChannelEnum.fromValue(rs.getString("channel"))).landArea(rs.getFloat("landArea"))
Index: rainmaker/pt-services-v2/src/main/java/org/egov/pt/web/models/Address.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- rainmaker/pt-services-v2/src/main/java/org/egov/pt/web/models/Address.java	(revision da92d8583591bd7d8ad9d44f3160a52eb865b80d)
+++ rainmaker/pt-services-v2/src/main/java/org/egov/pt/web/models/Address.java	(revision c21a11b74318b8a0fcdd610ac16dbca8a1f0146c)
@@ -10,6 +10,10 @@
 import lombok.NoArgsConstructor;
 import lombok.Setter;
 
+import javax.validation.Valid;
+import javax.validation.constraints.NotNull;
+import javax.validation.constraints.Size;
+
 /**
  * Address
  */
@@ -43,18 +47,23 @@
         @JsonProperty("type")
         private String type;
 
+        @NotNull
+        @Size(max=256)
         @JsonProperty("addressLine1")
         private String addressLine1;
 
+        @Size(max=256)
         @JsonProperty("addressLine2")
         private String addressLine2;
 
         @JsonProperty("landmark")
         private String landmark;
 
+        @NotNull
         @JsonProperty("city")
         private String city;
 
+
         @JsonProperty("pincode")
         private String pincode;
 
@@ -67,6 +76,7 @@
         @JsonProperty("street")
         private String street;
 
+        @Valid
         @JsonProperty("locality")
         private Boundary locality;
 
Index: rainmaker/pt-services-v2/src/main/java/org/egov/pt/web/models/Boundary.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- rainmaker/pt-services-v2/src/main/java/org/egov/pt/web/models/Boundary.java	(revision da92d8583591bd7d8ad9d44f3160a52eb865b80d)
+++ rainmaker/pt-services-v2/src/main/java/org/egov/pt/web/models/Boundary.java	(revision c21a11b74318b8a0fcdd610ac16dbca8a1f0146c)
@@ -4,6 +4,7 @@
 import java.util.List;
 
 import javax.validation.Valid;
+import javax.validation.constraints.NotNull;
 
 import org.springframework.validation.annotation.Validated;
 
@@ -27,6 +28,7 @@
 @NoArgsConstructor
 @Builder
 public class Boundary   {
+        @NotNull
         @JsonProperty("code")
         private String code;
 
Index: rainmaker/pt-services-v2/src/main/java/org/egov/pt/web/models/Property.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- rainmaker/pt-services-v2/src/main/java/org/egov/pt/web/models/Property.java	(revision da92d8583591bd7d8ad9d44f3160a52eb865b80d)
+++ rainmaker/pt-services-v2/src/main/java/org/egov/pt/web/models/Property.java	(revision c21a11b74318b8a0fcdd610ac16dbca8a1f0146c)
@@ -33,8 +33,6 @@
 @Setter
 @AllArgsConstructor
 @NoArgsConstructor
-//@Builder
-//@EqualsAndHashCode(of= {"id","tenantId"})
 public class Property extends PropertyInfo{
 
 
@@ -76,6 +74,7 @@
 	@JsonProperty("occupancyDate")
 	private Long occupancyDate;
 
+	@Valid
 	@JsonProperty("propertyDetails")
 	private List<PropertyDetail> propertyDetails;
 
@@ -176,94 +175,3 @@
 
 	}
 }
-
-
-
-
-/*	@JsonProperty("id")
-	private String id;
-
-	@JsonProperty("tenantId")
-	private String tenantId;
-
-	@JsonProperty("acknowldgementNumber")
-	private String acknowldgementNumber;
-
-	@JsonProperty("assessmentNumber")
-	private String assessmentNumber;*/
-
-	/**
-	 * status of the Property
-	 *//*
-	public enum StatusEnum {
-		ACTIVE("ACTIVE"),
-
-		INACTIVE("INACTIVE");
-
-		private String value;
-
-		StatusEnum(String value) {
-			this.value = value;
-		}
-
-		@Override
-		@JsonValue
-		public String toString() {
-			return String.valueOf(value);
-		}
-
-		@JsonCreator
-		public static StatusEnum fromValue(String text) {
-			for (StatusEnum b : StatusEnum.values()) {
-				if (String.valueOf(b.value).equals(text)) {
-					return b;
-				}
-			}
-			return null;
-		}
-	}
-
-	@JsonProperty("status")
-	private StatusEnum status;
-
-	@JsonProperty("financialYear")
-	private String financialYear;
-
-	@JsonProperty("propertyType")
-	private String propertyType;
-
-	@JsonProperty("propertySubType")
-	private String propertySubType;
-
-	@JsonProperty("usageCategoryMajor")
-	private String usageCategoryMajor;
-
-	@JsonProperty("owners")
-	@Valid
-	@NotNull
-	@Size(min=1)
-	private Set<OwnerInfo> owners;
-
-	@JsonProperty("address")
-	private Address address;*/
-
-	/*@JsonProperty("oldAssessmentNumber")
-	private String oldAssessmentNumber;
-
-	@JsonProperty("assessmentDate")
-	private Long assessmentDate;*/
-
-/**
- * New property comes into system either property is newly constructed or
- * existing property got sub divided. Here the reason for creation will be
- * captured.
- */
-
-/*
-	public Property addOwnersItem(OwnerInfo ownersItem) {
-		if (this.owners == null) {
-			this.owners = new HashSet<>();
-		}
-		this.owners.add(ownersItem);
-		return this;
-	}*/
Index: rainmaker/pt-services-v2/src/main/java/org/egov/pt/web/models/PropertyInfo.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- rainmaker/pt-services-v2/src/main/java/org/egov/pt/web/models/PropertyInfo.java	(revision da92d8583591bd7d8ad9d44f3160a52eb865b80d)
+++ rainmaker/pt-services-v2/src/main/java/org/egov/pt/web/models/PropertyInfo.java	(revision c21a11b74318b8a0fcdd610ac16dbca8a1f0146c)
@@ -74,6 +74,7 @@
         @JsonProperty("status")
         public StatusEnum status;
 
+        @Valid
         @JsonProperty("address")
         public Address address;
 
Index: rainmaker/pt-services-v2/src/main/resources/db/migration/main/V20180511150240__eg_pt_property_ddl.sql
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- rainmaker/pt-services-v2/src/main/resources/db/migration/main/V20180511150240__eg_pt_property_ddl.sql	(revision da92d8583591bd7d8ad9d44f3160a52eb865b80d)
+++ rainmaker/pt-services-v2/src/main/resources/db/migration/main/V20180511150240__eg_pt_property_ddl.sql	(revision c21a11b74318b8a0fcdd610ac16dbca8a1f0146c)
@@ -40,8 +40,7 @@
 
 
   CONSTRAINT pk_eg_pt_propertydetail_v2 PRIMARY KEY (financialYear,property),
-  CONSTRAINT uk_eg_pt_propertydetail_1_v2 UNIQUE (property),
-  CONSTRAINT uk_eg_pt_propertydetail_2_v2 UNIQUE (assessmentNumber),
+  CONSTRAINT uk_eg_pt_propertydetail_1_v2 UNIQUE (assessmentNumber),
   CONSTRAINT fk_eg_pt_propertydetail_v2 FOREIGN KEY (property) REFERENCES eg_pt_property_v2 (propertyId)
 );
 
Index: rainmaker/pt-services-v2/src/main/resources/property-services-v3.yml
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- rainmaker/pt-services-v2/src/main/resources/property-services-v3.yml	(revision da92d8583591bd7d8ad9d44f3160a52eb865b80d)
+++ rainmaker/pt-services-v2/src/main/resources/property-services-v3.yml	(revision c21a11b74318b8a0fcdd610ac16dbca8a1f0146c)
@@ -239,31 +239,6 @@
 
 
 
-        - query: INSERT INTO eg_pt_owner_v2(propertyDetail, userid, isActive, createdby, lastmodifiedby, createdtime, lastmodifiedtime) VALUES (?, ?, ?, ?, ?, ?, ?) ON CONFLICT (userid,propertyDetail) DO UPDATE SET isActive =?, lastmodifiedby =?, lastmodifiedtime =?;
-          basePath: properties.*.propertyDetails.*.owners.*
-          jsonMaps:
-
-           - jsonPath: $.properties.*.propertyDetails[*][?({id} in @.owners[*].id)].assessmentNumber
-
-           - jsonPath: $.properties.*.propertyDetails.*.owners.*.id
-
-           - jsonPath: $.properties.*.propertyDetails.*.owners.*.isActive
-
-           - jsonPath: $.properties[*][?({id} in @.propertyDetails.*.owners[*].id)].auditDetails.lastModifiedBy
-
-           - jsonPath: $.properties[*][?({id} in @.propertyDetails.*.owners[*].id)].auditDetails.lastModifiedBy
-
-           - jsonPath: $.properties[*][?({id} in @.propertyDetails.*.owners[*].id)].auditDetails.lastModifiedTime
-
-           - jsonPath: $.properties[*][?({id} in @.propertyDetails.*.owners[*].id)].auditDetails.lastModifiedTime
-
-           - jsonPath: $.properties.*.propertyDetails.*.owners.*.isActive
-
-           - jsonPath: $.properties[*][?({id} in @.propertyDetails.*.owners[*].id)].auditDetails.lastModifiedBy
-
-           - jsonPath: $.properties[*][?({id} in @.propertyDetails.*.owners[*].id)].auditDetails.lastModifiedTime
-
-
 
         - query: INSERT INTO eg_pt_propertydetail_v2(assessmentNumber,property,source,usage,noOfFloors,landArea,buildUpArea,additionalDetails,channel,financialYear,propertyType,propertySubType,usageCategoryMajor,assessmentDate,createdby, lastmodifiedby, createdtime,lastmodifiedtime) VALUES(?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?) ON CONFLICT ON CONSTRAINT pk_eg_pt_propertydetail_v2 DO UPDATE SET financialYear=?, propertyType=?, propertySubType=?,usageCategoryMajor=?, assessmentDate=?,source=?,usage=?,noOfFloors=?,landArea=?,buildUpArea=?,additionalDetails=?,channel=?,lastModifiedBy=?, lastModifiedTime=?;
           basePath: properties.*.propertyDetails.*
@@ -299,11 +274,11 @@
 
             - jsonPath: $.properties.*.propertyDetails.*.assessmentDate
 
-            - jsonPath: $.properties[*][?({assessmentNumber} in @.propertyDetails[*].assessmentNumber)].auditDetails.createdBy
+            - jsonPath: $.properties[*][?({assessmentNumber} in @.propertyDetails[*].assessmentNumber)].auditDetails.lastModifiedBy
 
             - jsonPath: $.properties[*][?({assessmentNumber} in @.propertyDetails[*].assessmentNumber)].auditDetails.lastModifiedBy
 
-            - jsonPath: $.properties[*][?({assessmentNumber} in @.propertyDetails[*].assessmentNumber)].auditDetails.createdTime
+            - jsonPath: $.properties[*][?({assessmentNumber} in @.propertyDetails[*].assessmentNumber)].auditDetails.lastModifiedTime
 
             - jsonPath: $.properties[*][?({assessmentNumber} in @.propertyDetails[*].assessmentNumber)].auditDetails.lastModifiedTime
 
@@ -368,6 +343,34 @@
             - jsonPath: $.properties[*][?({id} in @.propertyDetails.*.documents[*].id)].auditDetails.lastModifiedTime
 
 
+        - query: INSERT INTO eg_pt_owner_v2(propertyDetail, userid, isActive, createdby, lastmodifiedby, createdtime, lastmodifiedtime) VALUES (?, ?, ?, ?, ?, ?, ?) ON CONFLICT (userid,propertyDetail) DO UPDATE SET isActive =?, lastmodifiedby =?, lastmodifiedtime =?;
+          basePath: properties.*.propertyDetails.*.owners.*
+          jsonMaps:
+
+           - jsonPath: $.properties.*.propertyDetails[*][?({id} in @.owners[*].id)].assessmentNumber
+
+           - jsonPath: $.properties.*.propertyDetails.*.owners.*.id
+
+           - jsonPath: $.properties.*.propertyDetails.*.owners.*.isActive
+
+           - jsonPath: $.properties[*][?({id} in @.propertyDetails.*.owners[*].id)].auditDetails.lastModifiedBy
+
+           - jsonPath: $.properties[*][?({id} in @.propertyDetails.*.owners[*].id)].auditDetails.lastModifiedBy
+
+           - jsonPath: $.properties[*][?({id} in @.propertyDetails.*.owners[*].id)].auditDetails.lastModifiedTime
+
+           - jsonPath: $.properties[*][?({id} in @.propertyDetails.*.owners[*].id)].auditDetails.lastModifiedTime
+
+           - jsonPath: $.properties.*.propertyDetails.*.owners.*.isActive
+
+           - jsonPath: $.properties[*][?({id} in @.propertyDetails.*.owners[*].id)].auditDetails.lastModifiedBy
+
+           - jsonPath: $.properties[*][?({id} in @.propertyDetails.*.owners[*].id)].auditDetails.lastModifiedTime
+
+
+
+
+
         - query: INSERT INTO eg_pt_unit_v2(id,propertyDetail,floorNo,unitType,unitArea,usageCategoryMajor,usageCategoryMinor,usageCategorySubMinor,usageCategoryDetail,occupancyType,occupancyDate,constructionType,constructionSubType,arv,createdby, lastmodifiedby, createdtime, lastmodifiedtime) VALUES(?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?) ON CONFLICT (id) DO UPDATE SET floorNo=?,unitType=?,unitArea=?,usageCategoryMajor=?,usageCategoryMinor=?,usageCategorySubMinor=?,usageCategoryDetail=?,occupancyType=?,occupancyDate=?,constructionType=?,constructionSubType=?,arv=?,lastmodifiedby=?,lastmodifiedtime=?;
           basePath: properties.*.propertyDetails.*.units.*
           jsonMaps:
